src/widgets/helper/SearchPopup.cpp

```diff
@@ -88,17 +87,22 @@ void SearchPopup::addShortcuts()
         HotkeyCategory::PopupWindow, actions, this);
 }
 
-void SearchPopup::setChannelFilters(FilterSetPtr filters)
+void SearchPopup::addChannel(ChannelView &channel)
 {
-    this->channelFilters_ = std::move(filters);
-}
+    // TODO: user info card popups pajaW
 
-void SearchPopup::setChannel(const ChannelPtr &channel)
-{
-    this->channelView_->setSourceChannel(channel);
-    this->channelName_ = channel->getName();
-    this->snapshot_ = channel->getMessageSnapshot();
-    this->search();
+    if (this->searchChannels_.empty())
+    {
+        this->channelView_->setSourceChannel(channel.channel());
+        this->channelName_ = channel.channel()->getName();
+    }
+    else if (this->searchChannels_.size() == 1)
+    {
+        this->channelView_->setSourceChannel(
+            std::make_shared<Channel>("multichannel", Channel::Type::None));
+    }
+
+    this->searchChannels_.append(std::ref(channel));
```

Comment:
checking for duplicate channels here will avoid combining snapshots and iterating messages later on

Reply:
That would be nice, but the reason I'm not checking is you can have multiple splits of the same channel with different filters. I want to respect all filters when searching - not just the first split's.

Reply:
Also we'll still need to build this monolithic snapshot no matter what as each channel needs to be snapshotted in order to search it.

Reply:
didn't consider filters so make bigger sense now, waiting to test how it works out 👍